### ISSUE TITLE
@nrwl/react v10 build does not respect browserslist file (IE11 issue)

### ISSUE THREAD

**Issue by sushruth** [nrwl/nx#3422]
## Current Behavior

running `nx serve` or even `nx run build` without `--prod` flag seems to not respect browserslist configuration. 

This is probably a documentation issue. Having clarity on all steps needed to build for IE11 using different custom prod-like `configuration`s would be great.

## Expected Behavior

IE11 compatibility should be configurable.

## Steps to Reproduce

Create a new workspace -
```
❯ npx create-nx-workspace@latest
npx: installed 194 in 53.649s
? Workspace name (e.g., org name)     test
? What to create in the new workspace react             [a workspace with a single React application]
? Application name                    sample
? Default stylesheet format           CSS
? Use the free tier of the distributed cache provided by Nx Cloud? No  [Only use local computation cache]
Creating a sandbox with Nx...
[#########################################################################] 357/357
new test --preset="react" --appName="sample" --style="css" --interactive=false --collection=@nrwl/workspace
```
Repro repo - https://github.com/sushruth/nx-10-react-ie11-issue
built site -  https://nx-10-react-ie11-issue.sushruthsastry.vercel.app/

### Failure Logs

NA

### Environment

```
$ nx report

>  NX  Report complete - copy this into the issue template

  @nrwl/angular : Not Found
  @nrwl/cli : 10.0.6
  @nrwl/cypress : 10.0.6
  @nrwl/eslint-plugin-nx : 10.0.6
  @nrwl/express : Not Found
  @nrwl/jest : 10.0.6
  @nrwl/linter : 10.0.6
  @nrwl/nest : Not Found
  @nrwl/next : Not Found
  @nrwl/node : Not Found
  @nrwl/react : 10.0.6
  @nrwl/schematics : Not Found
  @nrwl/tao : 10.0.6
  @nrwl/web : 10.0.6
  @nrwl/workspace : 10.0.6
  typescript : 3.9.7
```

**Comment by thisiskylierose** [comment]
Can report this issue is occurring for our team as well. Any word on when we can get a fix?

**Comment by sushruth** [comment]
Referring to issue #2209 

Proposed a fix here - #3829
